cranelift/codegen/src/context.rs

```diff
@@ -159,18 +182,34 @@ impl Context {
 
         self.compute_domtree();
         self.eliminate_unreachable_code(isa)?;
-        if opt_level != OptLevel::None {
+
+        if !isa.flags().use_egraphs() && opt_level != OptLevel::None {
             self.dce(isa)?;
         }
 
         self.remove_constant_phis(isa)?;
 
-        if opt_level != OptLevel::None && isa.flags().enable_alias_analysis() {
+        if isa.flags().use_egraphs() {
+            log::debug!(
+                "About to optimize with egraph phase:\n{}",
+                self.func.display()
+            );
+            self.compute_domtree();
+            self.eliminate_unreachable_code(isa)?;
+            self.dce(isa)?;
+            self.remove_constant_phis(isa)?;
```

Comment:
Why is it necessary to re-run all of these passes after three of the four have just been run? Could you let the earlier DCE pass run if `isa.flags().use_egraphs() || opt_level != OptLevel::None` and then not need any of these?